src/object/cli_shard.c

```diff
@@ -423,17 +524,29 @@ dc_rw_cb(tse_task_t *task, void *arg)
 		if (rw_args->shard_args->auxi.flags & DRF_CHECK_EXISTENCE)
 			goto out;
 
+		is_ec_obj = (reasb_req != NULL) &&
+			    DAOS_OC_IS_EC(reasb_req->orr_oca);
+
 		if (rw_args->maps != NULL && orwo->orw_maps.ca_count > 0) {
+			daos_iom_t	*reply_maps;
+
 			/** Should have 1 map per iod */
 			D_ASSERT(orwo->orw_maps.ca_count == orw->orw_nr);
 			for (i = 0; i < orw->orw_nr; i++) {
-				daos_iom_copy(&orwo->orw_maps.ca_arrays[i],
-					      &rw_args->maps[i]);
+				reply_maps = &orwo->orw_maps.ca_arrays[i];
+				if (is_ec_obj &&
+				    reply_maps->iom_type == DAOS_IOD_ARRAY) {
+					obj_ec_iom_merge(reasb_req,
+							 orw->orw_tgt_idx,
+							 reply_maps,
+							 &rw_args->maps[i]);
```

Comment:
This probably should be done in object layer(obj_comp_cb() ?), instead of shard layer.

Reply:
now the replied iom possibly not accessible outside RPC's completion CB.

Reply:
I thought IOM  is from api argument?

Reply:
dest rw_args->maps is from API arg, src reply_maps is from RPC reply msg.

Reply:
I mean you copy these reply_maps into shard_auxi or somewhere per shard, then merge them inside object_comp_cb() ?

Reply:
I want to avoid extra mem alloc and copy for those reply_maps, it will be freed after RPC complete CB returns.

Reply:
Hmm, probably RPC should be freed in a different time. Anyway probably revisit later when cleanup the client stack.